src/pages/products/SingleProduct.jsx

```diff
@@ -0,0 +1,105 @@
+import { useState, useEffect } from 'react'
+import { Navbar } from 'components/navbar/Navbar.jsx'
+import { Footer } from 'components/footer/Footer.jsx'
+import { useParams, useNavigate, Link } from 'react-router-dom'
+import { products } from 'backend/db/products'
+import productStyle from './Products.module.css'
+import { useWishlistContext, useCart, useAuth } from 'context/index'
+import { containsInWishlist } from 'utils/wishlistUtils/index.jsx'
+import { wishlistHandler, addToCart } from 'services'
+function SingleProduct() {
+  let { productId } = useParams()
+  const navigate = useNavigate()
+  const [product, setProduct] = useState({})
+  const { wishlistItems, setWishlistItems } = useWishlistContext()
+  const { cartItems, cartDispatch } = useCart()
+  const { encodedToken } = useAuth()
+
+  useEffect(() => {
+    const product = products.find(product => product._id === productId)
+    product !== undefined ? setProduct(product) : navigate('/products')
+  }, [])
+  return (
+    <div className="home__container">
+      <Navbar />
+      <div className="main__container">
+        <div className={`text__lg ${productStyle['single-product']}`}>
+          <div className={productStyle['product-image']}>
+            <img src={product.img} alt={product.title} />
```

Comment:
you can destructure the product object further and use the property directly, and this was suggested by Tanay in one of our classes, you can do, const { img, title, ....} = product. and use the property directly